frugalos_segment/src/client/mod.rs

```diff
@@ -102,17 +102,30 @@ impl Client {
         };
         let object_id = id.clone();
         let logger = self.logger.clone();
-        self.mds
-            .put(id, metadata, expect, deadline, parent.clone())
-            .and_then(move |(version, created)| {
-                let mut tracking = PutFailureTracking::new(logger, object_id);
-                storage
-                    .put(version, content, deadline, parent)
-                    .map(move |()| {
-                        tracking.complete();
-                        (version, created)
-                    })
-            })
+
+        let mds = self.mds.clone();
+        let future = match expect {
+            Expect::Any => {
+                let f = mds
+                    .head(object_id.clone(), parent.clone())
+                    .map(move |version| version.map_or(Expect::None, |v| Expect::IfMatch(vec![v])));
```

Comment:
nit: この `move` はなくてもコンパイルが通ります。(何の所有権も奪っていないため)
```suggestion
                    .map(|version| version.map_or(Expect::None, |v| Expect::IfMatch(vec![v])));
```